Add ability to dynamically add/remove catalogs without restarting Presto (#24587)
prestodb/presto: presto-docs/src/main/sphinx/rest/catalog.rst

> =============

Reviewer: ```suggestion
================
```

> ===============

Reviewer: ```suggestion
================
```
The formatting line should exactly match the number of characters in the line being formatted. If it doesn't, a warning similar to this one appears in the doc build: 

```
/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/rest/catalog.rst:1: WARNING: Title overline too short.

===============
Catalog Resource
===============
```

Catalog Resource

> =============

Reviewer: ```suggestion
================
```

> ===============

Reviewer: ```suggestion
================
```

.. function:: GET /v1/catalog

   Returns a list of all catalogs currently registered.

   **Example response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Vary: Accept-Encoding, User-Agent
      Content-Type: application/json
      X-Content-Type-Options: nosniff
      Content-Length: 73

      ["system","tpch","tpcds","tpchstandard","hive","hive_bucketed","tpch_1"]

.. function:: POST /v1/catalog/{catalogName}

   Submits a catalog to register.

   **Example request**:

>     .. sourcecode:: http

Reviewer: This _Example request_ code example seems to be indented one space too much. Please un-indent each line in this code example (only) by one space. See this screenshot of a local doc build, the vertical gray stripe should not be present. 
All the other code blocks in this page are fine! It's just this one that needs fixing.

<img width="799" alt="Screenshot 2025-03-05 at 11 09 00 AM" src="https://github.com/user-attachments/assets/e0a1a396-e974-43b4-a097-e7fd6a68e07e" />

=============
Catalog Resource
=============

.. function:: GET /v1/catalog

>    Calling this service returns a list of all catalogs currently registered

Reviewer: ```suggestion
   Returns a list of all catalogs currently registered.
```
Suggest to parallel the other entries.

>    Will return a 403 if the request is not sent locally

Reviewer: ```suggestion
   Returns a 403 if the request is not sent locally.
```

   **Example response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Vary: Accept-Encoding, User-Agent
      Content-Type: application/json
      X-Content-Type-Options: nosniff
      Content-Length: 73

      ["system","tpch","tpcds","tpchstandard","hive","hive_bucketed","tpch_1"]

.. function:: POST /v1/catalog/{catalogName}

   Submits a catalog to register.

>    Will return a 403 if the request is not sent locally

Reviewer: ```suggestion
   Returns a 403 if the request is not sent locally.
```

   **Example request**:

    .. sourcecode:: http

      POST /v1/catalog/tpch_1 HTTP/1.1
      Host: 127.0.0.1:8080
      Accept: */*
      X-Presto-Prefix-Url: PREFIX
      X-Presto-User: PRESTO_USER
      User-Agent: StatementClient/0.55-SNAPSHOT
      Content-Type: application/json
      Content-Length: 88

      {
           "connector.name": "tpch",
           "tpch.splits-per-node": "4"
      }

   **Example response**:

   .. sourcecode:: http

      HTTP/1.1 201 Created
      Content-Type: application/json
      Content-Length: 0

.. function:: GET /v1/catalog/{catalogName}/status

   Returns the status of a requested catalog.

>    The possible result status values are ``catalog_in_use``, ``catalog_not_in_use``, ``catalog_not_found``

Reviewer: ```suggestion
   The possible result status values are ``catalog_in_use``, ``catalog_not_in_use``, ``catalog_not_found``.
```

>    Will return a 403 if the request is not sent locally

Reviewer: ```suggestion
   Returns a 403 if the request is not sent locally.
```

   **Example request**:

   .. sourcecode:: http

      GET /v1/catalog/system/status HTTP/1.1
      Host: 127.0.0.1:8080
      Accept: */*
      X-Presto-Prefix-Url: PREFIX
      X-Presto-User: PRESTO_USER
      User-Agent: StatementClient/0.55-SNAPSHOT
      Content-Type: application/json

   **Example response**:

   .. sourcecode:: http

      HTTP/1.1 200 OK
      Content-Type: application/json
      X-Content-Type-Options: nosniff
      Vary: Accept-Encoding, User-Agent

      {"message":"The given catalog system is currently in use","status":"catalog_in_use"}

.. function:: DELETE /v1/catalog/{catalogName}

>    Removes a catalog

Reviewer: ```suggestion
   Removes a catalog.
```

>    Will return a 403 if the request is not sent locally

Reviewer: ```suggestion
   Returns a 403 if the request is not sent locally.
```

   **Example request**:

   .. sourcecode:: http

      DELETE /v1/catalog/tpch_1 HTTP/1.1
      Host: 127.0.0.1:8080
      Accept: */*
      X-Presto-Prefix-Url: PREFIX
      X-Presto-User: PRESTO_USER
      User-Agent: StatementClient/0.55-SNAPSHOT
      Content-Type: application/json

   **Example response**:

   .. sourcecode:: http

      HTTP/1.1 204 No Content

.. function:: POST /v1/catalog/{catalogName}

   Updates a catalog with new parameters.

>    Possible response are 201, 204, 409

Reviewer: ```suggestion
   Possible responses are 201, 204, 409.
```

>    Will return a 403 if the request is not sent locally

Reviewer: ```suggestion
   Returns a 403 if the request is not sent locally.
```
